### PR TITLE
[EXPORTER] All 2xx return codes should be considered successful.

### LINKED ISSUE #2632: Is a 204 status code error something I should worry about when using opentelemetry-cpp.
I've built an application that sends telemetry data to a grafana cloud stack. The application sends both logs and traces. And I can query the traces and logs I send to grafana without any issues. The problem is though, when I send traces to grafana, I get a status code of 200, and opentelemetry-cpp logs a success message in cout. On the other hand for logs, I get a 204 status code which opentelemetry-cpp logs as an error message in cout. From my understanding of http status codes the 200 status codes are successes, plus my logs show up in the grafana data source, and I can query them without any issues. I was wondering if I could get confirmation on whether it's something I should worry about. And if it's something trivial, is there a way I can stop these logs from showing up in std::out.

**Describe your environment** 
I'm using windows 11. And My IDE of choice is visual studio.

**The log message in question**
```
[Debug] File: C:\.conan\e118c5\1\src\exporters\otlp\src\otlp_http_client.cc:175 [OTLP HTTP Client] Session state: session created
[Debug] File: C:\.conan\e118c5\1\src\exporters\otlp\src\otlp_http_client.cc:189 [OTLP HTTP Client] Session state: connecting to peer
[Debug] File: C:\.conan\e118c5\1\src\exporters\otlp\src\otlp_http_client.cc:750 [OTLP HTTP Client] Waiting for response from <Grafana OTLP Endpoint>/otlp/v1/logs (timeout = 10000 milliseconds)
[Debug] File: C:\.conan\e118c5\1\src\exporters\otlp\src\otlp_http_client.cc:207 [OTLP HTTP Client] Session state: connected
[Debug] File: C:\.conan\e118c5\1\src\exporters\otlp\src\otlp_http_client.cc:214 [OTLP HTTP Client] Session state: sending request
[Debug] File: C:\.conan\e118c5\1\src\exporters\otlp\src\otlp_http_client.cc:232 [OTLP HTTP Client] Session state: response received
[Error] File: C:\.conan\e118c5\1\src\exporters\otlp\src\otlp_http_client.cc:103 [OTLP HTTP Client] Export failed, Status:204, Header:   Date: Tue, 09 Apr,Body:08:30:05 GMT
[Error] File: C:\.conan\e118c5\1\src\exporters\otlp\src\otlp_http_log_record_exporter.cc:124 [OTLP HTTP Client] ERROR: Export 1 log(s) error: 1

```
**What is the expected behavior?**
I didn't expect it to be logged as an error.

**Additional context**
Non , but if there's any information you need from me, please ask.

### PR DESCRIPTION
**Pull request by owent**
Fixes #2632 

## Changes

According to https://datatracker.ietf.org/doc/html/rfc7231#page-47 . All 2xx return codes should be considered successful.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [x] Unit tests have been added
* [x] Changes in public API reviewed

### CODE REVIEW

File: exporters/zipkin/src/zipkin_exporter.cc [hunk 1]
```diff
@@ -78,7 +78,7 @@ sdk::common::ExportResult ZipkinExporter::Export(
   http_client::Body body_v(body_s.begin(), body_s.end());
   auto result = http_client_->PostNoSsl(url_parser_.url_, body_v, options_.headers);
   if (result &&
-      (result.GetResponse().GetStatusCode() == 200 || result.GetResponse().GetStatusCode() == 202))
+      (result.GetResponse().GetStatusCode() >= 200 || result.GetResponse().GetStatusCode() <= 299))
```
Comment (marcalff) [inline exporters/zipkin/src/zipkin_exporter.cc:81]: This will always be true.

Did you meant `(code >= 200) && (code <= 299)` ?

**Review by marcalff — COMMENTED** [review]
See issue in the zipkin exporter, errors are not detected.

File: exporters/zipkin/src/zipkin_exporter.cc (hunk 1 above)
Comment (owent) [reply exporters/zipkin/src/zipkin_exporter.cc]: Yes, and thanks.

**Review by marcalff — APPROVED** [review]
LGTM, thanks for the fix.

**Pull request merged**